sources/api/schnauzer/src/helpers/mod.rs

```diff
@@ -1426,7 +1524,18 @@ fn ecr_registry<S: AsRef<str>>(region: S) -> String {
     match partition {
         "aws-cn" => format!("{}.dkr.ecr.{}.amazonaws.com.cn", registry_id, region),
         "aws-iso-e" => format!("{}.dkr.ecr.{}.cloud.adc-e.uk", registry_id, region),
-        _ => format!("{}.dkr.ecr.{}.amazonaws.com", registry_id, region),
+        _ => format!(
+            "{}.dkr.ecr{}.{}.amazonaws.com",
+            registry_id,
+            // Only inject the fips service endpoint if the variant is in FIPS mode and the
+            // region supports FIPS.
+            if fips_enabled() && region.starts_with("us-") {
```

Comment:
Open to feedback that this is not robust enough and instead we should be checking against a set of the known FIPS-supported regions in all partitions.